### PR TITLE
GraphQL interface for products

### PR DESCRIPTION
**Pull request by pylipp**
new:
- GraphQL interface for products (mutations, input types, etc.): [files in `graphql/definitions/`](https://github.com/boxwise/boxtribute/pull/1266/files#diff-78d6e340f0803294242d845e5d92985a7bbd1b5d95d4dc6af0818053594750549)
- implementation of `createCustomProduct` mutation incl. error handling (using error result defined via GraphQL schema)

[Test plan](https://docs.google.com/spreadsheets/d/1sDhSsaVwNxAhGn1VYACDPVepZpvlwTkgy9nmeTaRy2Q/edit?pli=1#gid=1709672082) has been updated.

### Guideline for review
- start with `mutations.graphql` and check `inputs.graphql` and `types.graphql` as references
- the mutation resolver is implemented in `.../product/mutations.py`, check authorization there
- the resolver calls into a function in `.../product/crud.py`, check validation and creation of database entries (products and history table)
- error results that might occur according to the schema are defined in `errors.py` and tight to the schema in `bindables.py`

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0 ASSORT: Detailed GraphQL interface](https://trello.com/c/zGBPmp9d/1381-20-assort-detailed-graphql-interface)

File: back/boxtribute_server/graph_ql/definitions/protected/types.graphql [hunk 1]
```diff
@@ -524,6 +530,30 @@ type ShipmentDetail {
   receivedOn: Datetime
 }
 
+type InsufficientPermission {
```
Comment (HaGuesto) [inline back/boxtribute_server/graph_ql/definitions/protected/types.graphql:533]: Please make error return types visible via a naming convention

**Review by HaGuesto — APPROVED** [review]
We will see during frontend implemnetation if the errors key in the return json should be filled for errors or not.

File: back/boxtribute_server/graph_ql/definitions/protected/types.graphql (hunk 1 above)
Comment (pylipp) [reply back/boxtribute_server/graph_ql/definitions/protected/types.graphql:533]: https://github.com/boxwise/boxtribute/pull/1310/commits/a134a456a034f6f2697bf3d56c010f5ec4d386ee

**Pull request merged**